### PR TITLE
USB VID/PID: allow for boards to define default custom VID/PID

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

For boards that have terminals in their shipped configurations it makes sense to reuse those VID/PIDs when providing a terminal in RIOT as well.

That affects the builtin-peripherals IDs and not the custom specified ones.

Taken from https://github.com/RIOT-OS/RIOT/pull/13652#issuecomment-608378808

### Testing procedure

Set `RIOT_INTERNALPERIPHERAL_VID` and `RIOT_INTERNALPERIPHERAL_PID` for a board.
RIOT should use those IDs now and not spit out warnings when those IDs are used for stdio over USB.

### Issues/PRs references

alternative to #13652

### CODE REVIEW

**Comment by fjmolinas** [comment]
Can't this already be done by setting `CONFIG_USB_VID`? Why the need for the extra define?

**Comment by fjmolinas** [comment]
> Can't this already be done by setting CONFIG_USB_VID? Why the need for the extra define?

Sorry hadn't checked the discussion.

File: sys/include/usb.h [hunk 1]
```diff
@@ -24,17 +24,29 @@
...
+/* These can be overridden by boards that should come up with their board
+ * supplier VID/PID pair. Boards should only override this if the RIOT built-in
+ * peripherals are compatible with whatever is usually shipped with that pair
+ * */
+#ifndef RIOT_INTERNALPERIPHERAL_VID
+/** Reserved for RIOT standard peripherals as per http://pid.codes/1209/7D00/ */
+#define RIOT_INTERNALPERIPHERAL_VID (0x1209)
+/** Reserved for RIOT standard peripherals as per http://pid.codes/1209/7D00/ */
+#define RIOT_INTERNALPERIPHERAL_PID (0x7D00)
+#endif
```
Comment (fjmolinas) [inline sys/include/usb.h:44]:
```suggestion
#ifndef RIOT_INTERNALPERIPHERAL_VID
/** Reserved for RIOT standard peripherals as per http://pid.codes/1209/7D00/ */
#define RIOT_INTERNALPERIPHERAL_VID (0x1209)
#endif
#ifndef RIOT_INTERNALPERIPHERAL_PID
/** Reserved for RIOT standard peripherals as per http://pid.codes/1209/7D00/ */
#define RIOT_INTERNALPERIPHERAL_PID (0x7D00)
#endif
```

I'd rather have both guarded.

**Comment by fjmolinas** [comment]
> Set RIOT_INTERNALPERIPHERAL_VID and RIOT_INTERNALPERIPHERAL_PID for a board.
> RIOT should use those IDs now and not spit out warnings when those IDs are used for stdio over USB.

Are you talking about makefile warnings? Or other warnings? Makefile warnings are still present with this PR.

Otherwise I was able to override the definition on a samr21-xpro:

```
[10052.798514] cdc_acm 1-1.2.3:1.0: ttyACM1: USB ACM device
[10107.234803] usb 1-1.2.3: USB disconnect, device number 24
[10113.342776] usb 1-1.2.3: new full-speed USB device number 25 using xhci_hcd
[10113.445219] usb 1-1.2.3: New USB device found, idVendor=1309, idProduct=4400, bcdDevice= 0.00
[10113.445229] usb 1-1.2.3: New USB device strings: Mfr=3, Product=2, SerialNumber=0
[10113.445234] usb 1-1.2.3: Product: USB device
[10113.445239] usb 1-1.2.3: Manufacturer: RIOT-os.org
[10113.455652] cdc_acm 1-1.2.3:1.0: ttyACM1: USB ACM device
```

**Comment by benpicco** [comment]
> Are you talking about makefile warnings? Or other warnings? Makefile warnings are still present with this PR.

I only later found out that the warnings were actually caused by #14174, so this is just for cosmetics.

File: sys/include/usb.h [hunk 2]
```diff
@@ -24,17 +24,29 @@
...
 /**
  * @defgroup usb_conf USB peripheral compile time configurations
  * @ingroup config
  * @{
  */
 
+/* These can be overridden by boards that should come up with their board
+ * supplier VID/PID pair. Boards should only override this if the RIOT built-in
+ * peripherals are compatible with whatever is usually shipped with that pair
+ * */
```
Comment (fjmolinas) [inline sys/include/usb.h:38]: Nit-picking here but does the variable name make sense? If its overridne then its actually not `RIOT`s PID but the board or manufacturer right? Would it make more sense to just have `INTERNAL_PERIPHERAL_PID` (If my understanding of the variable meaning is correct)

**Comment by fjmolinas** [comment]
> I only later found out that the warnings were actually caused by #14174, so this is just for cosmetics.

Murdock warnings are fixed with https://github.com/RIOT-OS/RIOT/pull/14174, so this one is waiting for it.

**Comment by fjmolinas** [comment]
Warning should be fixed now.

**Comment by fjmolinas** [comment]
@benpicco do you agree with this PR as a prefered alternative to #13652?

**Comment by benpicco** [comment]
It touches less code, so that alone makes it preferable in my book.

**Comment by benpicco** [comment]
I also moved the configuration for `serpente` to the proper place now, looks like it's working:

```
[20299.302432] usb 2-1.5.3.1: new full-speed USB device number 21 using ehci-pci
[20299.413148] usb 2-1.5.3.1: New USB device found, idVendor=239a, idProduct=0057, bcdDevice= 0.00
[20299.413150] usb 2-1.5.3.1: New USB device strings: Mfr=3, Product=2, SerialNumber=0
[20299.413151] usb 2-1.5.3.1: Product: USB device
[20299.413152] usb 2-1.5.3.1: Manufacturer: RIOT-os.org
[20299.413741] cdc_acm 2-1.5.3.1:1.0: ttyACM3: USB ACM device
```

**Pull request merged**